src/meta/src/controller/catalog/alter_op.rs

```diff
@@ -782,4 +784,75 @@ impl CatalogController {
 
         Ok(version)
     }
+
+    // drop table associated source is a special case of drop relation, which just remove the source object and associated state table, keeping the streaming job and fragments.
+    pub async fn drop_table_associated_source(
+        txn: &DatabaseTransaction,
+        drop_table_connector_ctx: &DropTableConnectorContext,
+    ) -> MetaResult<(Vec<PbUserInfo>, Vec<PartialObject>)> {
+        {
+            // alter the original table to remove the ref to the state table catalog
+            let table_id = drop_table_connector_ctx.to_change_streaming_job_id;
+            Table::update(table::ActiveModel {
```

Comment:
Can we move it into `finish_replace_streaming_job_inner` to avoid update twice? This will make the function less prone to errors and its meaning more clear.